tests/testthat/setup-options.R

```diff
@@ -7,6 +14,7 @@ opts_partial_match_old <- list(
   warnPartialMatchAttr = getOption("warnPartialMatchAttr")
 )
 opts_partial_match_new <- list(
+  teal.lockfile.enable = FALSE,
```

Comment:
- please move it elsewhere (I mean outside of `opts_partial_match_new` as it has some meaning)
- coming back to [this](https://github.com/insightsengineering/teal/issues/1276#issuecomment-2288684906) - can we auto-dected that we are in the tests so that there is no need to set this option? This is relevant for tests outside of `teal` - e.g. `teal.modules.general`. I would like to avoid mass update all downstream packages to set this option in the tests - if possible of course.

Reply:
I think for tests you can detect if you are inside them with

```r

is_in_test <- function() {
  identical(Sys.getenv("TESTTHAT"), "true")
}
```

Reply:
R CMD check, for which we would also like to disable this option automatically

```r
is_r_cmd_check <- function() {
  ("CheckExEnv" %in% search()) || any(c("_R_CHECK_TIMINGS_", "_R_CHECK_LICENSE_") %in% names(Sys.getenv()))
}
```

Reply:
But using `is_in_test` and `is_r_cmd_check` is not ideal as we won't be able to test this functionality at all. Setting an option in tests in other packages allow us to test this functionality in `teal`

Reply:
> I think for tests you can detect if you are inside them with
> 
> ```r
> is_in_test <- function() {
>   identical(Sys.getenv("TESTTHAT"), "true")
> }
> ```

Have a look at `testthat::is_testing`.


> R CMD check, for which we would also like to disable this option automatically
> 
> ```r
> is_r_cmd_check <- function() {
>   ("CheckExEnv" %in% search()) || any(c("_R_CHECK_TIMINGS_", "_R_CHECK_LICENSE_") %in% names(Sys.getenv()))
> }
> ```

Checking for presence of env vars rather than their values may be risky because someone may choose to set them to `""` rather than unsetting them.

Reply:
Usage of `testthat::is_testing` would require putting `testthat` in Imports. The code that I put for `is_in_test` is actually the body of `testthat::is_testing`.

When it comes to those env vars - I hope they are created during R CMD CHECK and then they are unset. I don't think this is something a regular user would set.

Anyway I think this is not good practice to omit those in tests and R CMD CHECKS by design as there is no way to test this functionality. The option is better way, as can be turned off for any custom package, but not this one.

Reply:
Fair enough 👍  
Just to be clear, `is_testing` actually does `isTRUE(as.logical(Sys.getenv("TESTTHAT")))`.

Reply:
My testthat version does this

```r
> testthat::is_testing
function () 
{
    identical(Sys.getenv("TESTTHAT"), "true")
}
```

Reply:
My mistake.
```
> testthat:::env_var_is_true
function (x) 
{
    isTRUE(as.logical(Sys.getenv(x, "false")))
}
<bytecode: 0x558a1bf24230>
<environment: namespace:testthat>
```

Reply:
For other processes: `callr` sets `CALLR_IS_RUNNING` variable.


```r
library(callr)

process <- callr::r_bg(func = function() print(Sys.getenv()))
output <- process$read_output()
env <- strsplit(output, split = "\r\n")[[1]]
names_in_callr <- unique(gsub(" .*$", "", env))
setdiff(names_in_callr, names(Sys.getenv()))

Sys.getenv("CALLR_IS_RUNNING")
process <- callr::r_bg(func = function() print(Sys.getenv("CALLR_IS_RUNNING")))
output <- process$read_output()
output

[1] "CALLR_IS_RUNNING"                
[2] ""                                
[3] "CYGWIN"                          
[4] "PROCESSX_PSLQ3SBE3NYC_1726557642"
[5] "R_BROWSER"                       
[6] "R_PDFVIEWER"
```


`reprex` runs via `callr` : )

<img width="818" alt="image" src="https://github.com/user-attachments/assets/98e4ef04-9739-493f-a16e-7dff607b6486">

Reply:
@pawelru I've updated [a comment](https://github.com/insightsengineering/teal/issues/1276#issuecomment-2288684906). I think if we are about to disable lockfile generation in `callr` process then we have to do it always (not conditional on flag set in the parent process). If this is what you expect then we can continue

Reply:
If we want to disable lockfile generation for `callr`, `testthat::tests` and `R CMD CHECK` no matter what's the parent flag, then this needs to be added to `zzz.R -> onLoad`

```r
  lockfile_flag <-
    !(
      identical(Sys.getenv("CALLR_IS_RUNNING"), "true") || # inside callr process
      identical(Sys.getenv("TESTTHAT"), "true") || # inside devtools::test
      (
        ("CheckExEnv" %in% search()) || any(c("_R_CHECK_TIMINGS_", "_R_CHECK_LICENSE_") %in% names(Sys.getenv()))
      ) # inside R CMD CHECK
    )

  teal_default_options <- list(
    teal.show_js_log = FALSE,
    teal.lockfile.enable = lockfile_flag,
    shiny.sanitize.errors = FALSE
  )
```

I made a PR for this in here https://github.com/insightsengineering/teal/pull/1346

If we want to pay attention to the parent flag, then below code 
```r
  lockfile_flag <-
    !(
      identical(getOption("teal.lockfile.enable", ""), "") && # option is not set
      (
        identical(Sys.getenv("CALLR_IS_RUNNING"), "true") || # inside callr process
        identical(Sys.getenv("TESTTHAT"), "true") || # inside devtools::test
        (
          ("CheckExEnv" %in% search()) || any(c("_R_CHECK_TIMINGS_", "_R_CHECK_LICENSE_") %in% names(Sys.getenv()))
        ) # inside R CMD CHECK
      )
    )


  teal_default_options <- list(
    teal.show_js_log = FALSE,
    teal.lockfile.enable = lockfile_flag,
    shiny.sanitize.errors = FALSE
  )
```

Reply:
@pawelru this should be satisfied in this PR https://github.com/insightsengineering/teal/pull/1263/commits/761dd68e10426c451ba9eb1c0b8d2f40b0ead583

Reply:
resolving this and continuing the discussion in here https://github.com/insightsengineering/teal/pull/1263#discussion_r1766426661